lib/rdoc/parser/ruby.rb

```diff
@@ -1470,10 +1470,27 @@ def parse_method_params_and_body container, single, meth, added_container
         end
       end
 
+      coerce_call_seq(meth)
+
       parse_statements container, single, meth
     end
   end
 
+  # Coerce calling sequence if the given method is an operator method.
+  def coerce_call_seq(meth)
```

Comment:
This doesn't treat all operators correctly. For example, for the `!` method, it generates `self !` when it should be `!self`. It also doesn't handle the `[]` and `[]=` operators.

Reply:
`[]` and `[]=` (also `!`) are out of scope for this PR, which aims only at binary operators.

Reply:
The `~` method also doesn't accept any arguments.

Reply:
Fixed.